x-pack/plugins/ml/public/application/jobs/new_job/job_from_dashboard/quick_create_job_base.ts

```diff
@@ -196,15 +209,23 @@ export class QuickJobCreatorBase {
       this.kibanaConfig
     );
 
+    if (layerQuery) {
```

Comment:
For the layer query, I tried a few times, but it didn't look like the filter was getting passed to the datafeed when switching out to the Geo wizard from the flyout. The dashboard level filter was working ok. Does it work for you?

Reply:
I also see this.
Looking at the code, it appears the layer filters are not being added to the [page state](https://github.com/alvarezmelissa87/kibana/blob/b636a124c3b3b0a322eb0925ff270f78da5dd066/x-pack/plugins/ml/public/application/jobs/new_job/job_from_map/utils.ts#L25) for the redirect.

It is also not being extracted from the url [after the redirect ](https://github.com/alvarezmelissa87/kibana/blob/b636a124c3b3b0a322eb0925ff270f78da5dd066/x-pack/plugins/ml/public/application/jobs/new_job/job_from_map/route_resolver.ts#L17).

Reply:
Related to this, the map on the second and final steps of the Geo wizard looks like it is only using the data view and ignores any query / filter. Here for example I am creating a job on the ecommerce sample data from a saved search with `geoip.continent_name : "Europe"`:

<img width="1075" alt="image" src="https://user-images.githubusercontent.com/7405507/213435632-d901f1b5-8e94-4270-bb26-bcdebb92d38c.png">

The query is applied to the datafeed, but I think the maps should use it too.

Reply:
Good catch - passed along layer-level query when redirecting to the wizard in d2e4bdf704cbb8d740f7c048d9059875b5c8a4c3
cc @jgowdyelastic, @peteharverson 🙏

Reply:
I created an [issue](https://github.com/elastic/kibana/issues/149290) for the bug you found, @peteharverson, with the map not reflecting the active query/filter.